File: getting_started/insert_data.py
import csv

from schema import Address, Employee, Team
from terminusdb_client import WOQLClient

# we keep all the information in dictionaries with Employee id as keys
employees = {}
contact_numbers = {}
addresses = {}
managers = {}

with open("Contact.csv") as file:
    csv_file = csv.reader(file)
    next(csv_file)  # skiping header
    for row in csv_file:
        contact_numbers[row[0]] = row[1]
        street = row[2].split(",")[0]
        street_num = int(street.split(" ")[0])
        street_name = " ".join(street.split(" ")[1:])
        town = row[2].split(",")[1]
        addresses[row[0]] = Address(
            street_num=street_num, street=street_name, town=town, postcode=row[3]
        )

with open("Employees.csv") as file:
    csv_file = csv.reader(file)
    next(csv_file)  # skiping header
    for row in csv_file:
        team = eval(f"Team.{row[3].lower()}")
        employees[row[0]] = Employee(
            _id="Employee/" + row[0],
            name=row[1],
            title=row[2],
            address=addresses[row[0]],
            contact_number=contact_numbers[row[0]],
            team = team
        )
        managers[row[0]] = row[4]

for emp_id, man_id in managers.items():
    if man_id:
        employees[emp_id].manager = employees[man_id]

# For Terminus X, use the following
# client = WOQLClient("https://cloud.terminusdb.com/<Your Team>/")
# client.connect(db="demo_workshop", team="<Your Team>", use_token=True)

client = WOQLClient("http://127.0.0.1:6363/")
client.connect(db="getting_started")

client.insert_document(list(employees.values()), commit_msg="Adding 4 Employees")

File: getting_started/javascript-client/insert_data.js
const fs = require("fs");
const path = require("path");
const csv = require("fast-csv");

const TerminusClient = require("@terminusdb/terminusdb-client");


// TODO: Change teamname and username
const teamName = "yourTeam"
const username = "yourUser"

const client = new TerminusClient.WOQLClient(
  `https://cloud.terminusdb.com/${teamName}/`,
  { user: username, organization: teamName , db:"GettingStartedDB" }
);

//Assign your key to environment variable TERMINUSDB_ACCESS_TOKEN
client.setApiKey(process.env.TERMINUSDB_ACCESS_TOKEN);

const contact_numbers = {};
const addresses = {};
const employees = [];

// function to load and parse huge CSV files
const readCsv = (fileName) => {
    // TODO: change the directoryPath to point were the csv are stored
    return new Promise((resolve, reject) => {
      const data = [];
  
      fs.createReadStream(
        path.resolve(
          __dirname,
          fileName
        )
      )
        .pipe(csv.parse({ headers: true, ignoreEmpty: true }))
        .on("error", reject)
        .on("data", (row) => data.push(row))
        .on("end", () => {
          console.log(`Parsed ${data.length} rows`);
          resolve(data);
        });
    });
  };

const insertData = async () => {
  // read Contact.csv
  let resultContacts = await readCsv("Contact.csv");
  resultContacts.forEach((element) => {
    contact_numbers[element["Employee id"]] = element["Contact number"];

    let street = element["Home address"].split(",")[0];
    let street_num = Number(street.split(" ")[0]);
    let street_name = street.split(" ").slice(1).join(" ");
    let town = element["Home address"].split(",")[1].substr(1);

    addresses[element["Employee id"]] = {
      "@type": "Address",
      street: street_name,
      street_num,
      town,
      postcode: element["Postcode"],
    };
  });

  let resultEmployees = await readCsv("Employees.csv");
  resultEmployees.forEach((element) => {
    let employee = {
      "@type": "Employee",
      name: element["Name"],
      title: element["Title"],
      team: element["Team"],
      address: addresses[element["Employee id"]],
      contact_number: contact_numbers[element["Employee id"]],
      employee_id: element["Employee id"],
    };

    if (element["Manager"] !== "")
      employee.manager = "Employee/" + element["Manager"];

    employees.push(employee);
  });
  console.log("Inserting Employees ", employees);

  client
    .addDocument(employees)
    .then((res) => {
      console.log("Employees inserted successfully", res);
    })
    .catch((error) => {
      console.log(error);
    });

    const result = await client.getDocument({"as_list":true});
    console.log(result);
};

insertData();
